embedded-graphics/src/primitives/rounded_rectangle.rs

```diff
@@ -0,0 +1,778 @@
+///     .build();
+///
+/// RoundedRectangle::with_equal_corners(
+///     Rectangle::new(Point::new(5, 5), Size::new(50, 60)),
+///     Size::new(10, 10),
+/// )
+/// .into_styled(style)
+/// .draw(&mut display)?;
+/// # Ok::<(), core::convert::Infallible>(())
+/// ```
+///
+/// ## Different corner radii
+///
+/// This example creates a rounded rectangle 50px wide by 60px tall. Each corner is given a distinct
+/// radius in the x and y direction by creating a [`CornerRadii`](../struct.CornerRadiiBuilder.html)
+/// object and passing that to [`RoundedRectangle::new`](#method.new).
+///
+/// ```rust
+/// use embedded_graphics::{
+///     pixelcolor::Rgb565,
+///     prelude::*,
+///     primitives::{CornerRadii, Rectangle, RoundedRectangle},
+///     style::{PrimitiveStyle, PrimitiveStyleBuilder},
+/// };
+/// # use embedded_graphics::mock_display::MockDisplay;
+/// # let mut display = MockDisplay::default();
+///
+/// let style = PrimitiveStyleBuilder::new()
+///     .stroke_width(5)
+///     .stroke_color(Rgb565::RED)
+///     .fill_color(Rgb565::GREEN)
+///     .build();
+///
+/// let radii = CornerRadii {
+///     top_left: Size::new(10, 12),
+///     top_right: Size::new(14, 16),
+///     bottom_right: Size::new(18, 20),
+///     bottom_left: Size::new(22, 24),
+/// };
+///
+/// RoundedRectangle::new(Rectangle::new(Point::new(5, 5), Size::new(50, 60)), radii)
+///     .into_styled(style)
+///     .draw(&mut display)?;
+/// # Ok::<(), core::convert::Infallible>(())
+/// ```
+///
+/// ## Using `CornerRadiiBuilder`
+///
+/// This example creates a rounded rectangle 50px wide by 60px tall. Corner radii are set using the
+/// [`CornerRadiiBuilder`](../struct.CornerRadiiBuilder.html) builder.
+///
+/// ```rust
+/// use embedded_graphics::{
+///     pixelcolor::Rgb565,
+///     prelude::*,
+///     primitives::{CornerRadii, CornerRadiiBuilder, Rectangle, RoundedRectangle},
+///     style::{PrimitiveStyle, PrimitiveStyleBuilder},
+/// };
+/// # use embedded_graphics::mock_display::MockDisplay;
+/// # let mut display = MockDisplay::default();
+///
+/// let style = PrimitiveStyleBuilder::new()
+///     .stroke_width(5)
+///     .stroke_color(Rgb565::RED)
+///     .fill_color(Rgb565::GREEN)
+///     .build();
+///
+/// let radii = CornerRadiiBuilder::new()
+///     // Set the top left and top right corner radii to 10 x 20px
+///     .top(Size::new(10, 20))
+///     // Set the bottom right corner radius to 5 x 8px
+///     .bottom_right(Size::new(5, 8))
+///     .build();
+///
+/// RoundedRectangle::new(Rectangle::new(Point::new(5, 5), Size::new(50, 60)), radii)
+///     .into_styled(style)
+///     .draw(&mut display)?;
+/// # Ok::<(), core::convert::Infallible>(())
+/// ```
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
+pub struct RoundedRectangle {
+    /// The base rectangle
+    pub rectangle: Rectangle,
+
+    /// The radius of each corner
+    pub corners: CornerRadii,
+}
+
+impl RoundedRectangle {
```

Comment:
Should we add a `pub fn confine_radii(&self) -> RoundedRectangle` method, that returns a copy of the rectangle with confined radii?

That could be used if a user wants to implement some kind of algorithm that needs to know the "real" radii used by e-g.

Reply:
Good idea, added!